concent_api/core/views.py

```diff
@@ -261,123 +243,96 @@ def receive_out_of_band(request, _message):
             )
-            store_receive_out_of_band(golem_message, message_timestamp)
             message_verdict.sig = None
             return message_verdict
+        if last_undelivered_receive_status.type == 'RejectReportComputedTask':
+            serialized_reject_report_computed_task = last_undelivered_receive_status.data.tobytes()
 
-    if isinstance(decoded_last_task_message, message.RejectReportComputedTask):
-        if decoded_last_task_message.reason == message.RejectReportComputedTask.Reason.TASK_TIME_LIMIT_EXCEEDED:
-            rejected_task_id               = decoded_last_task_message.message_cannot_compute_task.task_to_compute.compute_task_def['task_id']
-            force_report_computed_task     = Message.objects.get(type = 'ForceReportComputedTask', task_id = rejected_task_id)
-            raw_force_report_computed_task = force_report_computed_task.data.tobytes()
-            try:
-                force_report_computed_task     = load(
-                    raw_force_report_computed_task,
-                    settings.CONCENT_PRIVATE_KEY,
-                    client_public_key
-                )
-            except InvalidSignature as exception:
-                return JsonResponse({'error': "Failed to decode ForceReportComputedTask. {}".format(exception)}, status = 400)
-
-            message_ack_report_computed_task.task_to_compute = force_report_computed_task.task_to_compute
-            message_verdict.ack_report_computed_task         = message_ack_report_computed_task
+            decoded_reject_report_computed_task = message.Message.deserialize(
+                serialized_reject_report_computed_task,
+                None,
+                check_time = False
+            )
 
-            last_undelivered_receive_out_of_band_status.delivered = True
-            last_undelivered_receive_out_of_band_status.full_clean()
-            last_undelivered_receive_out_of_band_status.save()
+            ack_report_computed_task                    = message.AckReportComputedTask()
+            ack_report_computed_task.task_to_compute    = decoded_reject_report_computed_task.task_to_compute
+            message_verdict.ack_report_computed_task    = ack_report_computed_task
 
             message_verdict.sig = None
             golem_message, message_timestamp = store_message(
                 type(message_verdict).__name__,
-                decoded_last_task_message.message_cannot_compute_task.task_to_compute.compute_task_def['task_id'],
+                decoded_reject_report_computed_task.cannot_compute_task.task_to_compute.compute_task_def['task_id'],
                 message_verdict.serialize()
             )
-            store_receive_out_of_band(golem_message, message_timestamp)
+            store_receive_out_of_band(
+                golem_message,
+                message_timestamp
+            )
             message_verdict.sig = None
             return message_verdict
+        return None
+    else:
```

Comment:
According to my previous comment about test I think this `else:` part can should be removed too.